File: chazm-api/src/main/java/runtimemodels/chazm/api/relation/Uses.java
package runtimemodels.chazm.api.relation;

import runtimemodels.chazm.api.entity.Pmf;
import runtimemodels.chazm.api.entity.Role;

/**
 * Represents the uses relation, where a {@linkplain Role} uses a {@linkplain Pmf}.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public interface Uses {
    /**
     * Returns {@linkplain Role} of this {@linkplain Uses}.
     *
     * @return {@linkplain Role} of this {@linkplain Uses}.
     */
    Role getRole();

    /**
     * Returns the {@linkplain Pmf} of this {@linkplain Uses}.
     *
     * @return the {@linkplain Pmf} of this {@linkplain Uses}.
     */
    Pmf getPmf();
}

File: chazm-api/src/main/kotlin/runtimemodels/chazm/api/relation/Uses.kt
package runtimemodels.chazm.api.relation

import runtimemodels.chazm.api.entity.Pmf
import runtimemodels.chazm.api.entity.Role
import runtimemodels.chazm.api.organization.Organization

/**
 * The [Uses] interface defines a uses relation, where a [Role] uses a [Pmf], of an [Organization].
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
interface Uses {
    /**
     * The [Role] of this [Uses].
     */
    val role: Role

    /**
     * The [Pmf] of this [Uses].
     */
    val pmf: Pmf
}
